frontend/src/lib/components/CommandBar/SearchResult.tsx

```diff
@@ -0,0 +1,86 @@
+import { useLayoutEffect, useRef } from 'react'
+import { useActions, useValues } from 'kea'
+
+import { resultTypeToName } from './constants'
+import { searchBarLogic, urlForResult } from './searchBarLogic'
+import { SearchResult as SearchResultType } from './types'
+import { LemonSkeleton } from '@posthog/lemon-ui'
+
+type SearchResultProps = {
+    result: SearchResultType
+    resultIndex: number
+    focused: boolean
+    keyboardFocused: boolean
+}
+
+const SearchResult = ({ result, resultIndex, focused, keyboardFocused }: SearchResultProps): JSX.Element => {
+    const { scrolling } = useValues(searchBarLogic)
+    const { onMouseEnterResult, onMouseLeaveResult, openResult, setScrolling } = useActions(searchBarLogic)
+
+    const ref = useRef<HTMLDivElement | null>(null)
+
+    useLayoutEffect(() => {
+        if (keyboardFocused) {
+            // :HACKY: This uses the non-standard scrollIntoViewIfNeeded api
+            // to improve scroll behaviour. Change to scrollIntoView({ scrollMode: 'if-needed' })
+            // once available.
+            if ((ref.current as any)?.scrollIntoViewIfNeeded) {
+                ;(ref.current as any).scrollIntoViewIfNeeded(false)
+            } else {
+                ref.current?.scrollIntoView()
+            }
+
+            // set scrolling state to prevent mouse enter/leave events during
+            // keyboard navigation
+            setScrolling(true)
+            setTimeout(() => {
```

Comment:
Will this trigger `setTimeout` multiple times? We should store the value and override / cancel it if triggered again

Reply:
Not sure if I'm following here. Do you mean a change of `keyboardFocused` from `false` -> `true` -> `false` -> `true` would trigger this twice?